docs/source/deployment/lambda.mdx

```diff
@@ -0,0 +1,387 @@
+
+If you ever want to remove the S3 bucket or Lambda functions that Serverless created, you can run the following command:
+
+```bash
+ serverless remove
+```
+
+## Customizing HTTP behavior
+
+The `@as-integrations/aws-lambda` package is compatible with Lambda's API Gateway V1 (REST) and V2 (HTTP). However, it does not let you customize HTTP behavior directly.
+
+If you'd like to customize your HTTP behavior, you can couple Apollo Server's Express integration (i.e., [`expressMiddleware`](../api/express-middleware)) with the [`@vendia/serverless-express`](https://github.com/vendia/serverless-express) package. The `@vendia/serverless-express` library translates between Lambda events and Express requests.
+
+You can update your Apollo Server setup to the following to have a fully functioning Lambda server that works in a variety of AWS features:
+
+<MultiCodeBlock>
+
+```ts
+const { ApolloServer } = require('@apollo/server');
+const { expressMiddleware } = require('@apollo/server/express4');
+const serverlessExpress = require('@vendia/serverless-express');
+const express = require('express');
+const { json } = require('body-parser');
+const cors = require('cors');
+
+const server = new ApolloServer({
+  typeDefs: 'type Query { x: ID }',
+  resolvers: { Query: { x: () => 'hi!' } },
+});
+
+server.startInBackgroundHandlingStartupErrorsByLoggingAndFailingAllRequests();
+
+const app = express();
+app.use(cors(), json(), expressMiddleware(server));
+
+exports.graphqlHandler = serverlessExpress({ app });
+```
+
+</MultiCodeBlock>
+
+The setup enables you to customize your HTTP behavior as needed.
+
+## Using request information
+
+You can use the [`context` function](../data/resolvers/#the-context-argument) to get information about the current operation from the original Lambda data structures. Your `context` function can access this information from the passed-in `event` and `context` arguments:
+
+<MultiCodeBlock>
+
+```ts
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+
+// This final export is important!
+export const graphqlHandler = startServerAndCreateLambdaHandler(server, {
+  // highlight-start
+  context: async (event, context) => {
+    return {
+      lambdaEvent: event,
+      lambdaContext: context,
+    };
+  },
+  // highlight-end
+});
+```
+
+</MultiCodeBlock>
+
+The `event` object contains the API Gateway event (HTTP headers, HTTP method, body, path, etc.). The `context` object (not to be confused with the `context` function) contains the current Lambda Context (function name, function version, `awsRequestId`, time remaining, etc.).
+
+If you've [changed your set up to use `@vendia/serverless-express`](#customizing-http-behavior) you'll also have access to an `express` argument, containing the Express request and response in your `context` function:
```

Comment:
I'm not sure what you mean by "an `express` argument" here.

If you're using this setup then the context function receives Express-style information as its argument, and the Lambda stuff is available via calling getCurrentInvoke. There's no argument named `express`.

Reply:
I forgot to update that sentence, thank you! I changed it to:

> If you've [changed your setup to use `@vendia/serverless-express`](#customizing-http-behavior) your `context` function receives `req` and `res` options which are [`express.Request`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2c4c1d0b106a1305c384cde84bb14bbf1a938d2c/types/express-serve-static-core/index.d.ts#L367) and [`express.Response`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2c4c1d0b106a1305c384cde84bb14bbf1a938d2c/types/express-serve-static-core/index.d.ts#L659) objects:

As well as updated the code snippet with a callout for where folks have access to the API Gateway and Lambda context:
```ts
app.use(
  cors(),
  json(),
  expressMiddleware(server, {
    // The Express request and response objects are passed into
    // your context initialization function
    context: async ({ req, res }) => {
      // Here is where you'll have access to the
      // API Gateway event and Lambda Context
      const { event, context } = serverlessExpress.getCurrentInvoke();
      return {
        expressRequest: req,
        expressResponse: res,
        lambdaEvent: event,
        lambdaContext: context,
      };
    },
  }),
);
```